numbat/modules/core/mixed_units.nbt

```diff
@@ -15,3 +17,14 @@ fn _mixed_unit_list<D: Dim>(val: D, units: List<D>, acc: List<D>) -> List<D> =
     if (len(units) > 0)
       then (val |> trunc_in(head(units)))
       else error("Units list cannot be empty")
+  
+
+fn _clean_units<D: Dim>(units: List<D>) -> List<D> = units |> unique() |> sort_descending()
+
+fn _unit_list<D: Dim>(units: List<D>, value: D) -> List<D> = _mixed_unit_list(value, _clean_units(units), [])
+
+@description("Round a mixed unit number to the nearest multiple of `base`.")
+@example("[0 days, 12 hours, 0 minutes, 30 seconds] |> round_mixed_in(minutes)")
+@example("[0 days, 12 hours, 29 minutes, 30 seconds] |> round_mixed_in(hours)")
+fn round_mixed_in<D: Dim>(base: D, value: List<D>) -> List<D> = value |> sum |> round_in(base) |> _unit_list(units)
+  where units = value |> filter(is_not_zero) |> map(unit_of) |> cons(base)
```

Comment:
Maybe
```suggestion
fn round_mixed_in<D: Dim>(base: D, value: List<D>) -> List<D> =
  value |> sum |> round_in(base) |> _unit_list(units)
    where units = value |> filter(is_not_zero) |> map(unit_of) |> cons(base)
```

Reply:
This function seems quite special. Maybe move it to wherever you need it and then keep it private for now?

Reply:
If so, maybe remove it from this PR and add it later? Same for sort_descending.

Reply:
Sure, I'll do the same as for sort_descending.